mofa/simulation/lammps.py

```diff
@@ -8,9 +14,103 @@ class LAMMPSRunner:
+        Args:
+            cif_path: starting structure's cif file path
+            timesteps: Number of timesteps to run
+            report_frequency: How often to report structures
+        Returns:
+            lmp_path: a directory with the lammps simulation input files
+            return_code: cif2lammps running status, 0 means success (directory lmp_path will be kept), -1 means failure (directory lmp_path will be destroyed)
+        """
+        cif_name = os.path.split(cif_path)[-1]
+        lmp_path = os.path.join(self.lmp_sims_root_path, cif_name.replace(".cif", ""))
+        os.makedirs(lmp_path, exist_ok=True)
+        try:
+            single_conversion(cif_path,
+                              force_field=UFF4MOF,
+                              ff_string='UFF4MOF',
+                              small_molecule_force_field=None,
+                              outdir=lmp_path,
+                              charges=False,
+                              parallel=False,
+                              replication='2x2x2',
+                              read_cifs_pymatgen=True,
+                              add_molecule=None,
+                              small_molecule_file=None)
+            in_file_name = [x for x in os.listdir(lmp_path) if x.startswith("in.") and not x.startswith("in.lmp")][0]
+            data_file_name = [x for x in os.listdir(lmp_path) if x.startswith("data.") and not x.startswith("data.lmp")][0]
+            in_file_rename = "in.lmp"
+            data_file_rename = "data.lmp"
+            print("Reading data file for element list: " + os.path.join(lmp_path, data_file_name))
+            with io.open(os.path.join(lmp_path, data_file_name), "r") as rf:
+                df = pd.read_csv(io.StringIO(rf.read().split("Masses")[1].split("Pair Coeffs")[0]), sep=r"\s+", header=None)
+                element_list = df[3].to_list()
+            with io.open(os.path.join(lmp_path, in_file_rename), "w") as wf:
+                print("Writing input file: " + os.path.join(lmp_path, in_file_rename))
+                with io.open(os.path.join(lmp_path, in_file_name), "r") as rf:
+                    print("Reading original input file: " + os.path.join(lmp_path, in_file_name))
+                    wf.write(rf.read().replace(data_file_name, data_file_rename) + """
+
+# simulation
+
+fix                 fxnpt all npt temp 300.0 300.0 $(200.0*dt) tri 1.0 1.0 $(800.0*dt)
+variable            Nevery equal """ + "%d" % report_frequency + """
+
+thermo              ${Nevery}
+thermo_style        custom step cpu dt time temp press pe ke etotal density xlo ylo zlo cella cellb cellc cellalpha cellbeta cellgamma
+thermo_modify       flush yes
+
+minimize            1.0e-10 1.0e-10 10000 100000
+reset_timestep      0
+
+dump                trajectAll all custom ${Nevery} dump.lammpstrj.all id type element x y z q
+dump_modify         trajectAll element """ + " ".join(element_list) + """
```

Comment:
Are you familiar with the f-string syntax in Python? Instead of the summations, you could just do
```python
f"""
...
dump_modify traj element {" ".join(element_list)}
"""

Don't feel obligated to change this, but consider it for the future.
```